### ISSUE TITLE
Unable to run kubetest2 <deployer> --test more than once from same path

### ISSUE THREAD

**Issue by Rajalakshmi-Girish** [kubernetes-sigs/kubetest2#176]
**Summary:** We are unable to use kubetest2 to test kubernetes more than once from a same path.

`metadata.json` generated by kubetest2 command is objecting to run `kubetest2` second time from same path stating the following error:
Example using gingko tester:
```
root@kubetest2:/workspace# kubetest2 tf --test=ginkgo -- --flake-attempts 2 --focus-regex='ariable Expansion should verify that a failing subpath expansion can be modified during the lifecycle of a container'
F1210 11:00:28.745660   14208 ginkgo.go:205] failed to run ginkgo tester: key tester-version already exists in the metadata
Error: exit status 255
root@kubetest2:/workspace#
```

**Expected result:** Should be able to run kubetest2 any number of times from a given path. There should ether be a flag to ignore metadata.json or overwrite the metadata with latest kubetest2 run

In order to run k8s conformance tests by following option B at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#running-conformance-tests-with-kubetest, we are required to trigger kubetest2 command twice. 
```
failed to run ginkgo tester: key tester-version already exists in the metadata
Error: exit status 255
```
is the error thrown when run from same path.

**Note:** This started happening after the below commit to fail if key is already present in metadata.json.
https://github.com/kubernetes-sigs/kubetest2/pull/162/commits/ac36e3203cb49dce808f5bc211e1decc63956d93
Expect this check to be more liberal https://github.com/kubernetes-sigs/kubetest2/blob/master/pkg/metadata/metadata.go#L49

**Sample job** that threw this error: https://prow.ppc64le-cloud.org/view/gs/ppc64le-kubernetes/logs/test-periodic-kubernetes-conformance-test-ppc64le/1468899640811196416

**Comment by Rajalakshmi-Girish** [comment]
@mkumatag ^^

**Comment by BenTheElder** [comment]
fixed by your pending PR https://github.com/kubernetes-sigs/kubetest2/pull/183, right?

**Comment by Rajalakshmi-Girish** [comment]
> fixed by your pending PR #183, right?

This seems to be fixed by https://github.com/kubernetes-sigs/kubetest2/pull/162
Closing this issue as we no longer encounter this after upgrading to latest `kubetest2`

**Comment by Rajalakshmi-Girish** [comment]
This error started coming back after moving metadata.json from RunDir to ARTIFACTS via  https://github.com/kubernetes-sigs/kubetest2/pull/221

IMO, if a second `kubetest2` command is not allowed to run from the same path as the first `kubetest2`, The error message has to suggest what can be done.
- Like changing the ARTIFACTS directory to a different one
OR
-  Maybe even allow overwriting of `tester-version` key in metadata.json if `kubetest2` is run a second time from same path.
@BenTheElder your thoughts on this, please.

**Comment by BenTheElder** [comment]
IMHO: metadata.json should be part of ARTIFACTS, it's expected to be uploaded in CI and read by the CI tools.

We should probably just overwrite or amend the existing file.

In the meantime, kubetest2 should probably be used with unique $ARTIFACTS, maybe we should instead use a tempdir for the ARTIFACTS location if it's not explicitly set.

I've stepped down to emeritus_approver in this project though, I switched roles / transferred teams earlier this year and this was on my "I am still doing too many things" list.
